### DISCUSSION TITLE
Support PHP8 named arguments for custom twig functions / variables

### DISCUSSION THREAD

**Discussion by mandrasch** [craftcms/cms#13576] [Ideas]
Hey,

I have a custom twig function / var, which I would love to call with named arguments (since I have a lot more arguments than the example ;-))

Tried:

```
{% set entries = craft.myFilterUtils.filter(
    siteId=currentSite.id,
    anotherParam='foo'
) %}
```

Result: `Arguments must be separated by a comma. Unexpected token "operator" of value "=" ("punctuation" expected with value ",")`

Custom twig variable was created in a module via 

```php
// Module code:

// Expose Vars to Twig in module
Event::on(CraftVariable::class, CraftVariable::EVENT_INIT, function (Event $event) {
    /** @var CraftVariable $variable */
    $variable = $event->sender;
    $variable->set('filterUtils', FilterUtilsVariables::class);
});

// ...

// Class:

class FilterUtilsVariables
{

  public function filter(
        int     $siteId,
        ?string $param1 = null,
        ?string $anotherParam = null,
        // ... much more params
  )
}
```

Based on: https://craftcms.com/docs/4.x/extend/extending-twig.html

It seems to work for twig builtins like `include()` according to https://craftquest.io/livestreams/named-arguments-in-twig though. 

Thanks in advance, best regards!

**Comment by brandonkelly** [comment]
Twig has supported named arguments since before PHP 8 had them, but it only works for functions/filters added via extensions; not for methods of global variables.

**Reply by mandrasch** [reply]
Thanks very much for quick reply, @brandonkelly!

(I'll try with https://craftcms.com/docs/4.x/extend/extending-twig.html#register-a-twig-extension next 👍 )
